platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/streams/SerializableDataOutputStream.java

```diff
@@ -295,4 +286,22 @@ protected void writeClassIdVersion(final SerializableDet serializable, final boo
         }
         this.writeInt(serializable.getVersion());
     }
+
+    /**
+     * Write a PBJ record to the stream
+     *
+     * @param record
+     * 		the record to write
+     * @param codec
+     * 		the codec to use to write the record
+     * @param <T>
+     * 		the type of the record
+     * @throws IOException
+     * 		thrown if any IO problems occur
+     */
+    public <T extends Record> void writePbjRecord(@NonNull final T record, @NonNull final Codec<T> codec)
+            throws IOException {
+        writeInt(codec.measureRecord(record));
+        codec.write(record, writableSequentialData);
```

Comment:
The length is known, so a check can be added how many bytes have been actually written and compared against the length. A similar check is there when objects are read from input streams.

Reply:
I would expect that codec would throw if it can't write the whole record, is that not the case?

Reply:
Such checks (there are a lot in other parts of the repo) are to catch bugs in codecs, when `measureRecord()` and `write()` are not consistent with each other. It's better get an exception on writing than to debug why reading fails much later.